src/utils/span-pointers.ts

```diff
@@ -0,0 +1,74 @@
+import { eventTypes } from "../trace/trigger";
+import { logDebug } from "./log";
+import {
+  SPAN_LINK_KIND,
+  S3_PTR_KIND,
+  SPAN_POINTER_DIRECTION,
+  generateS3PointerHash,
+} from "dd-trace/packages/dd-trace/src/span_pointers";
+
+interface SpanPointerAttributes {
+  "ptr.kind": string;
+  "ptr.dir": string;
+  "ptr.hash": string;
+  "link.kind": string;
+}
+
+/**
+ * Computes span pointer attributes
+ *
+ * @param {eventTypes} eventSource - The type of event being processed (e.g., S3, DynamoDB).
+ * @param {any} event - The event object containing source-specific data.
+ * @returns {SpanPointerAttributes[] | undefined} An array of span pointer attribute objects, or undefined if none could be computed.
+ */
+export function getSpanPointerAttributes(
+  eventSource: eventTypes | undefined,
+  event: any,
+): SpanPointerAttributes[] | undefined {
+  if (!eventSource) {
+    return;
+  }
+
+  switch (eventSource) {
+    case eventTypes.s3:
+      return processS3Event(event);
+    default:
+      logDebug(`Event type ${eventSource} not supported by span pointers.`);
+      return;
+  }
+}
+
+function processS3Event(event: any): SpanPointerAttributes[] {
+  const records = event.Records || [];
+  const spanPointerAttributesList = [];
+  const linkKind = SPAN_LINK_KIND;
+
+  for (const record of records) {
+    const eventName = record.eventName;
+    if (!["ObjectCreated:Put", "ObjectCreated:Copy", "ObjectCreated:CompleteMultipartUpload"].includes(eventName)) {
```

Comment:
why restrict us to only these particular object created types instead of `ObjectCreated:*`?

Reply:
We only support creating span pointers for these three operations on the upstream case. I guess it doesn't hurt to create span pointers downstream in other cases; we could change the check to

`if (!eventName.startsWith("ObjectCreated"))`